File: Frontend/src/js/model/account.js
export class Account {
  accountId;
  accountNumber;
  accountNameVi;
  accountNameEn;
  parentId;
  parentNumber;
  categoryKind;
  categoryKindName;
  description;
  isParent;
  grade;
  mCodeId;
  detailByBankAccount;
  detailByAccountObject;
  detailByAccountObjectKind;
  foreignCurrencyAccounting;
  usingStatus;
  usingStatusName;
  constructor(a) {
    this.accountId = a.accountId ?? "";
    this.accountNumber = a.accountNumber ?? "";
    this.accountNameVi = a.accountNameVi ?? "";
    this.accountNameEn = a.accountNameEn ?? "";
    this.parentId = a.parentId ?? "";
    this.parentNumber = a.parentNumber ?? "";
    this.categoryKind = a.categoryKind != null ? a.categoryKind.toString() : "";
    this.categoryKindName = a.categoryKindName ?? "";
    this.description = a.description ?? "";
    this.isParent = a.isParent ?? false;
    this.grade = a.grade ?? 0;
    this.mCodeId = a.mCodeId ?? "";
    this.detailByBankAccount = a.detailByBankAccount;
    this.detailByAccountObject = a.detailByAccountObject;
    this.detailByAccountObjectKind = a.detailByAccountObjectKind;
    this.foreignCurrencyAccounting = a.foreignCurrencyAccounting ?? false;
    this.usingStatus = a.usingStatus ?? true;
    this.usingStatusName = a.usingStatusName ?? "Đang sử dụng";
  }

  cloneFromOtherAccount(a) {
    this.accountId = a.accountId;
    this.accountNumber = a.accountNumber;
    this.accountNameVi = a.accountNameVi;
    this.accountNameEn = a.accountNameEn;
    this.parentId = a.parentId;
    this.parentNumber = a.parentNumber;
    this.categoryKind = a.categoryKind;
    this.categoryKindName = a.categoryKindName;
    this.description = a.description;
    this.isParent = a.isParent;
    this.grade = a.grade;
    this.mCodeId = a.mCodeId;
    this.detailByBankAccount = a.detailByBankAccount;
    this.detailByAccountObject = a.detailByAccountObject;
    this.detailByAccountObjectKind = a.detailByAccountObjectKind;
    this.foreignCurrencyAccounting = a.foreignCurrencyAccounting;
    this.usingStatus = a.usingStatus;
    this.usingStatusName = a.usingStatusName;
  }

  convertToApiFormat() {
    const obj = {
      accountNumber: this.accountNumber,
      accountNameVi: this.accountNameVi,
      accountNameEn: this.accountNameEn,
      parentId: this.parentId != "" ? this.parentId : null,
      parentNumber: this.parentNumber,
      categoryKind: this.categoryKind,
      categoryKindName: this.categoryKindName,
      description: this.description,
      detailByBankAccount: this.detailByBankAccount,
      detailByAccountObject: this.detailByAccountObject,
      detailByAccountObjectKind: this.detailByAccountObjectKind,
      foreignCurrencyAccounting: this.foreignCurrencyAccounting,
      usingStatus: this.usingStatus,
      usingStatusName: this.usingStatusName,
    };
    return obj;
  }
}

File: Backend/MISA.WebFresher032023.Demo.Repository/Entities/Output/Account.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MISA.WebFresher032023.Demo.DataLayer.Entities.Output
{
    public class Account
    {
        public Guid AccountId { get; set; }
        public string AccountNumber { get; set; }
        public string AccountNameVi { get; set; }
        public string? AccountNameEn { get; set; }
        public bool IsParent { get; set; }
        public Guid? ParentId { get; set; }
        public string? ParentNumber { get; set; }
        public int CategoryKind { get; set; }
        public string CategoryKindName { get; set; }
        public string? Description { get; set; }
        public bool? DetailByBankAccount { get; set; }
        public bool? DetailByAccountObject { get; set; }
        public int? DetailByAccountObjectKind { get; set; }
        public int Grade { get; set; }
        public string MCodeId { get; set; }
        public bool? ForeignCurrencyAccounting { get; set; }
        public bool? UsingStatus { get; set; }
        public string UsingStatusName { get; set; }
    }
}
